src/main/java/emu/grasscutter/game/player/Player.java

```diff
@@ -1187,6 +1189,7 @@ public void onLogin() {
         session.send(new PacketAllWidgetDataNotify(this));
         session.send(new PacketWidgetGadgetAllDataNotify());
         session.send(new PacketCombineDataNotify(this.unlockedCombines));
+        session.send(new PacketGetChatEmojiCollectionRsp(this.getChatEmojiIdList()));
```

Comment:
Rsp packets should only be sent as a response to the corresponding Req packets. You already send GetChatEmojiCollectionRsp via HandlerGetChatEmojiCollectionReq::handle, so this is redundant at best
```suggestion

```

Reply:
you are right, but client doesn't seem to send request packet.

Reply:
Then the issue is elsewhere. Perhaps the client expects some prior notification packet from the server for it to then send the Req packet (as is the case with ActivityScheduleInfoNotify and GetActivityInfoReq). Rsp packets with no prior Req shouldn't happen.

Reply:
I will try to find the prior packet.

Reply:
I noticed that `PacketPlayerLoginRsp` wasn't sent.